KIALI-1201 ServiceDetails: Move DestinationRules into a Table

aljesusg opened the pull request:
Required : #530 [KIALI-1220](https://issues.jboss.org/browse/KIALI-1220)

JIRA: [KIALI-1201](https://issues.jboss.org/browse/KIALI-1201)

### Before
<img width="1303" alt="captura de pantalla 2018-07-22 a las 18 36 10" src="https://user-images.githubusercontent.com/3019213/43047958-13885502-8de0-11e8-93f4-c856142da9be.png">

### After

<img width="1862" alt="captura de pantalla 2018-07-22 a las 18 48 23" src="https://user-images.githubusercontent.com/3019213/43047957-135dfe42-8de0-11e8-9e9b-88b9613c6deb.png">

pilhuhn commented:
I think I would move the created at/source version to the right and the 
policy/subsets in their current place, as they are much more important 
than especially the resource version.

aljesusg commented:
Moved, thanks @pilhuhn

xeviknal commented:
@aljesusg I guess this table is an intermediate step to move DR's details to another page, right? Otherwise, I would remove subsets and traffic policy from the table.
However, there is this row title repetition '[host]' or 'subsets' that I think there are redundant. What do you think?

aljesusg commented:
Changed target to istio-1.0
Removed redundant.
![screenshot from 2018-07-30 09-13-32](https://user-images.githubusercontent.com/3019213/43382718-fa6df9ec-93d8-11e8-9d3c-894d342d60b0.png)

abonas commented:
@kiali/ux-reviewers fyi. any comments?

cshinn commented:
@aljesusg @abonas do subsets ever have any traits besides a name and a set of labels? If not, I think it might be clearer (and more space efficient) to remove the word "labels" put the name inline with any associated labels (wrapping to a second line as needed) 

Maybe something like this

![group 14](https://user-images.githubusercontent.com/3300756/43402729-68e6ed4a-93e1-11e8-9f0c-e1a0c0ed8e9b.png)

abonas commented:
@cshinn yes, subsets might have also trafficPolicy optional section.
The structure is as follows: name and labels are required (there could be more than 1 label) and trafficPolicy is optional - if it exists, it overrides a "higher level" traffic policy
https://istio.io/docs/reference/config/istio.networking.v1alpha3/#Subset

@aljesusg Why is the name of the subset appearing after the labels and with [] brackets - is that intentional? 
I'd also make it bold and reverse the order of labels and name, unless, again there is some logic for this particular order that I've missed.

aljesusg commented:
Yes @abonas it's heredate from the component and how we had the information, we can change it.

aljesusg commented:
Changed @cshinn  @abonas . thanks
![screenshot from 2018-07-31 11-33-45](https://user-images.githubusercontent.com/3019213/43451689-cdb093b4-94b5-11e8-973f-55eb987255e5.png)

abonas commented:
It seems that the names are removed now? Is this intentional?

aljesusg commented:
yes @abonas because it has no sense I think maybe @xeviknal or @jotak can tell us why is name field in subsets
```json
subsets: [
  {
     labels: {
        version:"v1"
     },
     name: "v1"
   },
   {
     labels: {
         version:"v2"
     },
     name: "v2"
    },
...
]
```

abonas commented:
@aljesusg this information is coming from Istio, so I don't think it should be removed in Kiali. I think the reason for naming the subsets is to help distinguishing them more easily.

aljesusg commented:
Ok.Thanks @abonas Changed it
![screenshot from 2018-07-31 13-08-46](https://user-images.githubusercontent.com/3019213/43456276-220a49f2-94c3-11e8-801b-270b6618085c.png)

abonas commented:
@aljesusg great :)
Do you have an example how it would look with trafficPolicy? (an optional field for a subset)

abonas commented:
@aljesusg isn't this an optional property of each subset separately?

jotak commented:
@aljesusg the git history seems wrong here. Is it supposed to go in master branch or in istio-1.0 ? There's a lot of commits in that PR that are obviously not related to your task. Can you fix it? (I think the simplest would be to reset to upstream master and cherry-pick your commit.)

jotak commented:
@aljesusg  @abonas for what I can see, traffic policy can be set both at destination rule's root and in subsets. So I guess on the screenshot above, it is set at rule's root, not on subset.
How does it look like with a rule such as:

```
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: bookinfo-ratings
spec:
  host: ratings.prod.svc.cluster.local
  trafficPolicy:
    loadBalancer:
      simple: LEAST_CONN
  subsets:
  - name: testversion
    labels:
      version: v3
    trafficPolicy:
      loadBalancer:
        simple: ROUND_ROBIN
```

(taken from istio doc https://istio.io/docs/reference/config/istio.networking.v1alpha3/ )

aljesusg commented:
Fixed @jotak it seems that I get the last commits when I changed the base branch. Thanks !!!

aljesusg commented:
@jotak Is the backend showing the traffic policy of each subset or we must do it?

jotak commented:
@aljesusg I have no idea, it's more @xeviknal 's part I guess :)

xeviknal commented on src/pages/ServiceDetails/ServiceInfo/ServiceInfoDestinationRules.tsx:
```diff
@@ -14,42 +16,155 @@ class ServiceInfoDestinationRules extends React.Component<ServiceInfoDestination
+    );
+  }
+
+  generateSubsets(subsets: Subset[]) {
```
Here, as @jotak mention, is missing the displaying of a trafficPolicy for each subset.
If you see in the definition of [Subset](https://istio.io/docs/reference/config/istio.networking.v1alpha3/#Subset), it can option have a trafficPolicy.
For adding it, I would use the `<DetailObject name="trafficPolicy" detail={subset.trafficPolicy} />`.
